Content.Server/Speech/EntitySystems/MeleeSpeechSystem.cs

```diff
@@ -1,49 +1,85 @@
-
+using Robust.Server.GameObjects;
+using Robust.Shared.Players;
 
 namespace Content.Server.Speech.EntitySystems;
 
 public sealed class MeleeSpeechSystem : SharedMeleeSpeechSystem
 {
 
-	[Dependency] private readonly IAdminLogManager _adminLogger = default!;
-
-
-	public override void Initialize()
+    [Dependency] private readonly IAdminLogManager _adminLogger = default!;
+    [Dependency] private readonly SharedActionsSystem _actionSystem = default!;
+    [Dependency] private readonly UserInterfaceSystem _uiSystem = default!;
+    public override void Initialize()
 	{
 		base.Initialize();
 		SubscribeLocalEvent<MeleeSpeechComponent, MeleeSpeechBattlecryChangedMessage>(OnBattlecryChanged);
-	}
-
+        SubscribeLocalEvent<MeleeSpeechComponent, MeleeSpeechConfigureActionEvent>(OnConfigureAction);
+        SubscribeLocalEvent<MeleeSpeechComponent, GetItemActionsEvent>(OnGetActions);
+        SubscribeLocalEvent<MeleeSpeechComponent, ComponentInit>(OnComponentInit);
+    }
+
+    private void OnComponentInit(EntityUid uid, MeleeSpeechComponent component, ComponentInit args)
+    {
+        if (component.ConfigureAction != null)
+            _actionSystem.AddAction(uid, component.ConfigureAction, uid);
+    }
+    
+    private void OnGetActions(EntityUid uid, MeleeSpeechComponent component, GetItemActionsEvent args)
+    {
+        if (component.ConfigureAction != null)
+            args.Actions.Add(component.ConfigureAction);
+    }
 
 	private void OnBattlecryChanged(EntityUid uid, MeleeSpeechComponent comp, MeleeSpeechBattlecryChangedMessage args)
 	{
 
         if (!TryComp<MeleeSpeechComponent>(uid, out var meleeSpeechUser))
-			return;
+            return;
 
-        string battlecry = args.Battlecry;
+        var battlecry = args.Battlecry;
 
         if (battlecry.Length > comp.MaxBattlecryLength)
             battlecry = battlecry[..comp.MaxBattlecryLength];
 
         TryChangeBattlecry(uid, battlecry, meleeSpeechUser);
 	}
+    /// <summary>
+    /// Attempts to open the Battlecry UI.
+    /// </summary>
+    private void OnConfigureAction(EntityUid uid, MeleeSpeechComponent comp, MeleeSpeechConfigureActionEvent args)
+    {
+
+        TryOpenUi(args.Performer, uid, comp);
+    }
 
 
+    public void TryOpenUi(EntityUid user, EntityUid storeEnt, MeleeSpeechComponent? component = null)
+    {
+        if (!Resolve(storeEnt, ref component))
+            return;
 
+        if (!TryComp<ActorComponent>(user, out var actor))
+            return;
 
-	/// <summary>
-	/// Attempts to change the battlecry of an entity.
-	/// Returns true/false.
-	/// </summary>
-	/// <remarks>
-	/// If provided with a player's EntityUid to the player parameter, adds the change to the admin logs.
-	/// </remarks>
-	public bool TryChangeBattlecry(EntityUid uid, string? battlecry, MeleeSpeechComponent? meleeSpeechComp = null)
+        if (!_uiSystem.TryToggleUi(storeEnt, MeleeSpeechUiKey.Key, actor.PlayerSession))
+            return;
```

Comment:
```suggestion
        _uiSystem.TryToggleUi(storeEnt, MeleeSpeechUiKey.Key, actor.PlayerSession);
```

no need to wrap it in a condition